README.md

```diff
@@ -1,23 +1,1377 @@
+We always prefer _more_ detailed instructions when learning
+so we have added more detail to each step.
+Crucially we know all the steps in _this_ tutorial _work_ flawlessly,
+because the counter works in the finished example.
+If you followed the instructions in "Step 0"
+to run the finished app on your `localhost`
+_before_ diving into building it,
+you also know they work for _you_. ✅
+
+
+Open the `mix.exs` file and locate the `deps` list,
+e.g:
+
+```elixir
+defp deps do
+  [
+    {:phoenix, "~> 1.4.15"},
+    {:phoenix_pubsub, "~> 1.1"},
+    {:phoenix_html, "~> 2.11"},
+    {:phoenix_live_reload, "~> 1.2", only: :dev},
+    {:gettext, "~> 0.11"},
+    {:jason, "~> 1.0"},
+    {:plug_cowboy, "~> 2.0"}
+  ]
+end
+
+```
+
+Append the following line to the end of the list:
+
+```
+{:phoenix_live_view, "~> 0.8.1"},
+```
+
+The `deps` definition should now look likes this:
+
+```elixir
+defp deps do
+  [
+    {:phoenix, "~> 1.4.15"},
+    {:phoenix_pubsub, "~> 1.1"},
+    {:phoenix_html, "~> 2.11"},
+    {:phoenix_live_reload, "~> 1.2", only: :dev},
+    {:gettext, "~> 0.11"},
+    {:jason, "~> 1.0"},
+    {:plug_cowboy, "~> 2.0"},
+    {:phoenix_live_view, "~> 0.8.1"},
+  ]
+end
+```
+
+The _last_ line in the code block is the important one.
+
+
+> 🏁 [`mix.exs`](https://github.com/dwyl/phoenix-liveview-counter-tutorial/blob/79d149c89655a6ddd452c93187e638b487aaf375/mix.exs#L33-L46)
+file at the end of Step 2:
+[mix.exs#L44](https://github.com/dwyl/phoenix-liveview-counter-tutorial/blob/79d149c89655a6ddd452c93187e638b487aaf375/mix.exs#L44)
+
+<br />
+
+#### 2.1 Download the `phoenix_live_view` Dependency
+
+Now that you've added the `phoenix_live_view` to `mix.exs`,
+you need to _download_ the dependencies.
+Run:
+
+```
+mix deps.get
+```
+
+You should see output similar to:
+
+```
+Resolving Hex dependencies...
+Dependency resolution completed:
+Unchanged:
+  cowboy 2.7.0
+  ...
+New:
+  phoenix_live_view 0.8.1
+* Getting phoenix_live_view (Hex package)
+```
+
+<br />
+
+### Step 3. Configure `signing_salt` in `config.exs`
+
+Phoenix LiveView uses a cryptographic salt
+to secure communications
+between client and server. 🔐 <br />
+You don't need to know what this is,
+just follow the instructions below and you'll be fine,
+but if you are curious,
+read: https://en.wikipedia.org/wiki/Salt_(cryptography)
+
+In your terminal run the following command:
+
+```
+mix phx.gen.secret 32
+```
+
+You should see output similar to the following:
+
+```
+iluKTpVJp8PgtRHYv1LSItNuQ1bLdR7c
+```
+
+> 💡 This is a **random** string generator
+that generates a **32 character string** of alphanumeric data, <br />
+so the result will be **different each time** you run the command.
+
+Copy the string into your computer's clipboard.
+
+
+Open your `config/config.exs` file
+and locate the line that begins with
```

Comment:
I'm not sure the configuration `live_view` is added by default when the live_view dependency is installed.
In my case I had to add (not just replace) the value.

Reply:
Ah, please ensure you're using Phoenix `1.4.15` 💭

Reply:
ok I'm using 1.4.11

Reply:
Either way I've added a section instructing people to add: 
```elixir
config :live_view_counter, LiveViewCounterWeb.Endpoint,
   live_view: [
     signing_salt: "iluKTpVJp8PgtRHYv1LSItNuQ1bLdR7c"
   ]
```
to their `config.exs` if they don't _already_ have a `live_view` config. 👍